src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -432,6 +434,22 @@ def get_role(self, name):
                 raise RoleDoesNotExist(name)
             return self.roles[name]
 
+    def increment_attempt(self, username, success=False):
+        with self.lock:
+            if username in self.users:
+                if success:
+                    attempt = 0
+                else:
+                    attempt = self.get_attempt(username) + 1
+                self.users[username].attempt = attempt
+
+    def get_attempt(self, username):
+        with self.lock:
```

Comment:
You can use `setdefault` to achieve the same with less code.
```suggestion
        with self.lock:
        	return self.users[username].setdefault('attempt', 0)
```

Reply:
`setdefault` works for a dictionary, but here the users[username] is an object of the class Users, so I think `setdefault` doesnt work over here. @votdev

Reply:
Sorry, i currently see the `KeyError` is addressing the `self.users[username]`. Argghhhh, my fault.

Reply:
No problem. I didn't even knew we can do this for dictionary. Thank you for that.